python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -1197,27 +1245,117 @@ async def set_hexapod(self, azimuth, elevation, axis=f'xyzuvw'):
+                self.model.add_offset("z", _offset_value)
+                self.focus_offset_yet_to_be_applied -= _offset_value
+                # publish events with new offsets
+                self.evt_correctionOffsets.set_put(**self.model.offset,
+                                                   force_output=True)
+                # Do accounting to republish total offset and others that were
+                # set in the callbacks
+                self.focus_offset_per_category['total'] = self.model.offset['z']
+                self.evt_focusOffsetSummary.set_put(**self.focus_offset_per_category,
+                                                    force_output=True)
+                await asyncio.sleep(0)
+
+            # Now do pointing
+            if abs(np.max(self.pointing_offsets_yet_to_be_applied)) > abs(_pointing_tolerance):
+                self.log.info(
+                    f'Applying pointing offset of [X,Y]={self.pointing_offsets_yet_to_be_applied} '
+                    'in correction loop due to filter '
+                    'and/or disperser changes.')
+                _pointing_offsets = copy.deepcopy(self.pointing_offsets_yet_to_be_applied)
+            elif(_offset_value):
+                # No offsets above thresholds, so just return
+                self.log.debug('Focus and pointing offsets below tolerances. Passing without correcting')
+                pass
+
+            # Corrections required, so flip the bit on the detailed state
+            self.detailed_state = self.detailed_state ^ status_bit
 
-    def update_mount_position_callback(self, id_data):
-        """Callback function to update the position of the telescope.
+            self.log.debug(f'_offset_value is {_offset_value}')
+            self.log.debug(f'_pointing_offsets is {_pointing_offsets!r}')
 
-        Parameters
-        ----------
-        id_data : SALPY_ATMCS.ATMCS_mountEncoders
+            # send out even saying correction is started
+            self.evt_atspectrographCorrectionStarted.set_put(focusOffset=_offset_value,
+                                                             pointingOffsets=_pointing_offsets)
 
-        """
-        self.azimuth = id_data.azimuthCalculatedAngle[-1]
-        self.elevation = id_data.elevationCalculatedAngle[-1]
+            try:
+                # Don't need a tolerance since these values only get set if
+                # they meet a tolerance already
+                if abs(_offset_value):
+                    self.log.debug('Applying focus correction with hexapod')
+                    await self.set_hexapod(azimuth, elevation)
+
+                if np.max(abs(_pointing_offsets)):
+                    self.log.debug('Applying pointing correction _pointing_offsets'
+                                   f' = {_pointing_offsets} with atcs')
+                    # apply offsets relative to what is already there
+                    await self.atcs.offset_xy(_pointing_offsets[0], _pointing_offsets[1],
```

Comment:
I'm adding a `persistent` option to `offset_xy` that will make sure the offset persist between slews. Can you make sure you pass `persistent=True` here?